packages/db/src/translate/collections.ts

```diff
@@ -1,6 +1,37 @@
-import { NewCollectionRecord } from '@cumulus/types/api/collections';
-import { PostgresCollection } from '../types/collection';
+import { NewCollectionRecord, CollectionRecord } from '@cumulus/types/api/collections';
+import { PostgresCollection, PostgresCollectionRecord } from '../types/collection';
+const { removeNilProperties } = require('@cumulus/common/util');
 
+/**
+* Translates a PostgresCollectionRecord object to a `NewCollectionRecord` API collection object
+* @param {PostgresCollectionRecord} collectionRecord - PostgreSQL collection record to translate
+* @returns {CollectionRecord} - Translated record
+*/
+export const translatePostgresCollectionToApiCollection = (
+  collectionRecord: PostgresCollectionRecord
+): CollectionRecord => removeNilProperties(({
+  createdAt: collectionRecord.created_at.getTime(),
+  updatedAt: collectionRecord.updated_at.getTime(),
+  name: collectionRecord.name,
+  version: collectionRecord.version,
+  process: collectionRecord.process,
+  url_path: collectionRecord.url_path,
+  duplicateHandling: collectionRecord.duplicate_handling,
+  granuleId: collectionRecord.granule_id_validation_regex,
+  granuleIdExtraction: collectionRecord.granule_id_extraction_regex,
+  files: collectionRecord.files,
+  reportToEms: collectionRecord.report_to_ems,
+  sampleFileName: collectionRecord.sample_file_name,
+  ignoreFilesConfigForDiscovery: collectionRecord.ignore_files_config_for_discovery,
+  meta: collectionRecord.meta ?? undefined,
```

Comment:
Interesting - is this passing TS validation? Because `meta` on `PostgresCollectionRecord` is a `string`, since the actual value is a JSON string. But on `CollectionRecord` it is an `Object`, since we expect a JSON object. So I'm confused how assigning the PostgresCollectionRecord `meta` directly to the `meta` for `CollectionRecord` is working

Reply:
Oh dear.  That would be because the api type is using `Object` instead of `object`.    string is an `Object` but not an `object`

Reply:
Very nice catch.

Reply:
I think 05183a1fb should sort it.      I'll see if we  can do something about the linting situation during the IP.

Reply:
2628f3b3a actually.    So what's happening here is this: 

When we write a stringified meta object to the database it gets translated TO jsob.    When we *read* it via knex, the *object* comes back.  

As such, this line *is* correct and it *is* an `object` (but *NOT* an `Object`), e.g.: 

```
> record = await cpg.get(knex, {})
{
  cumulus_id: 1,
  name: '4d89f',
  version: '0.0.0',
  sample_file_name: 'file.txt',
  granule_id_validation_regex: 'fake-regex',
  granule_id_extraction_regex: 'fake-regex',
  files: [ { regex: 'fake-regex', sampleFileName: 'file.txt' } ],
  process: null,
  url_path: null,
  duplicate_handling: null,
  report_to_ems: null,
  ignore_files_config_for_discovery: null,
  meta: { foo: 'bar' },
  tags: null,
  created_at: 2021-06-28T18:15:49.893Z,
  updated_at: 2021-06-28T18:15:49.893Z
}
> record.meta
{ foo: 'bar' }
```

Reply:
That all looks good, but if `meta` and `tags` are actual objects when returned by Knex from the database, should we update these properties in the `PostgresCollectionRecord` type to be `object`?

```
  meta?: string,
  tags?: string,
```

Reply:
I'm still confused how this is passing TS compilation without the change I'm suggesting to `PostgresCollectionRecord`. 

Because based on the types at least, it thinks `collectionRecord.meta` is a `string` and we're mapping it directly to `meta` of `ApiCollection` which is now a type of `object`

Reply:
For meta, absolutely.   For tags, however, more like: f027ff901